### PR TITLE
docs: add dev setup guide

### LINKED ISSUE #55: Add developer setup instructions in Contributing Guidelines
Add instructions for potential contributors to setup the package for local development with aws-toolkit-vscode. Not having any documentation on this was quite frustrating for me and isn't friendly for newcomers who might want to start contributing.

### PR DESCRIPTION
**Pull request by yaythomas**
Add developer setup instructions to `CONTRIBUTING.md`, for first time contributors. Include setup for hot-reloading and debugging locally with `aws-toolkit-vscode`.

Closes #55.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by abdelnn — APPROVED** [review]

**Review by evscott — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,6 +43,69 @@ GitHub provides additional document on [forking a repository](https://help.githu
 ## Finding contributions to work on
 Looking at the existing issues is a great way to find something to contribute on. As our projects, by default, use the default GitHub issue labels (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any 'help wanted' issues is a great place to start.
 
+## Getting Started
+### Setup
+To work on this project you need these environmental dependencies:
+-   [NodeJS and NPM](https://nodejs.org/) (latest version of both)
+-   [Git](https://git-scm.com/downloads)
+-   [AWS `git secrets`](https://github.com/awslabs/git-secrets)
```
Comment (wong-a) [inline CONTRIBUTING.md:51]: This seems useful, but it isn't integrated in this package nor mentioned in the rest of this doc is it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,6 +43,69 @@ GitHub provides additional document on [forking a repository](https://help.githu
...
+# make current project directory available for linking
+npm link
+```
+
+2) In your `aws-toolkit-vscode` repo root, run:
+```
+# link current project to the local language service
+npm link amazon-states-language-service
+
+# optional. verify amazon-states-language-service dependency pointing to your
```
Comment (wong-a) [inline CONTRIBUTING.md:83]: nit: Would move the comments out of the code block into regular text

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -43,6 +43,69 @@ GitHub provides additional document on [forking a repository](https://help.githu
...
+Then clone the repository and install the project dependencies with NPM:
+
+```
+git clone git@github.com:aws/amazon-states-language-service.git
+cd amazon-states-language-service
+npm install
+npm test
+```
+
+### Debug
```
Comment (wong-a) [inline CONTRIBUTING.md:62]: nit: "Linking with AWS Toolkit for VS Code" would be a more accurate heading. Debugging instructions (setting up a debugger) is mentioned in the `Run` section

**Review by wong-a — COMMENTED** [review]
Thanks for updating the setup instructions! 🚀

File: CONTRIBUTING.md (hunk 1 above)
Comment (yaythomas) [reply CONTRIBUTING.md:51]: Resolved in #98

File: CONTRIBUTING.md (hunk 3 above)
Comment (yaythomas) [reply CONTRIBUTING.md:62]: resolved in #98

**Pull request merged**